### PR TITLE
add integrity checks to `IntervalHeap`

### PR DESCRIPTION
**Pull request by apasel422**
cc #85

### CODE REVIEW

File: src/interval_heap.rs [hunk 1]
```diff
@@ -399,7 +407,6 @@ mod test {
             let mut ih = IntervalHeap::from_vec(tmp);
             for _ in range(0, 100) {
                 ih.push(rng.next_u32());
-                assert!(is_interval_heap(as_slice(&ih)));
```
Comment (Gankro) [inline src/interval_heap.rs]: Are debug asserts guaranteed to run in `cargo test`?

Comment (tbu-) [inline src/interval_heap.rs]: Yes, they are. `cargo test` is built in non-release mode.

Comment (apasel422) [inline src/interval_heap.rs]: Good question. Per [crates.io] (http://doc.crates.io/manifest.html#the-[profile.*]-sections), the test profile uses `cfg(ndebug)` by default. It's possible to disable in `Cargo.toml`, I suppose, but I don't know what the rationale would be, and I doubt you'd merge such a change to the project.

File: src/interval_heap.rs [hunk 2]
```diff
@@ -338,6 +351,21 @@ impl<T, C: Compare<T>> IntervalHeap<T, C> {
     pub fn clear(&mut self) {
         self.data.clear();
     }
+
+    fn is_valid(&self) -> bool {
```
Comment (Gankro) [inline src/interval_heap.rs]: Now that this isn't just "test" code, I'd appreciate some documentation of this.

Comment (apasel422) [inline src/interval_heap.rs]: Will do. I can also try to write it more idiomatically.

File: src/interval_heap.rs (hunk 1 above)
Comment (apasel422) [inline src/interval_heap.rs]: `collect`'s `Cargo.toml` actually _has_ debugging disabled as of 202076e1. I will change that now that the relevant issue has been fixed in Rust master.

**Comment by apasel422** [comment]
@Gankro Added some documentation to `is_valid`, tried to make the code more readable, and added a test.

**Comment by Gankro** [comment]
Nice! This looks good to me.

**Pull request merged**